governance/ApiDeprecation.md

```diff
@@ -0,0 +1,116 @@
+# API Deprecation
+
+> Unsure about what type of deprecation you're facing? Please see this page for more information - [Deprecation Meta](DeprecationMeta.md)
+
+> Wondering if this applies to you? This process only applies to public APIs - if you're sure consumers of your API will not be impacted, making an immediate change is okay.
+
+## Overview
+
+Examples of a API deprecation:
+
+- Renaming a function
+- Renaming a class/EBus
+- Changing a function signature
+- Removing a function
+- Removing a class/EBus
+- Changing the address type of an EBus
+- Changing the preconditions of a function
+- e.g. Function no longer handles nullptrs or parameter values must now fall within a specified range (e.g. between `0.0` and `1.0`).
+
+## Process
+
+### Golden Rule
+
+> Do not make any breaking changes to an API when first deprecating it (this includes adding the `AZ_DEPRECATED` macro). The code must work exactly as before.
+
+### Steps (Guide)
+
+1. Identify the code to be deprecated (something resembling one of the API deprecation examples outlined in the Overview).
+1. Create a new GitHub issue with the name "Deprecate <Function/Type Name>"
+    1. Add the Deprecation label.
+    1. In the Description, add a preliminary Release Note (what do you want to say to the customer about this change).
+    1. Fill in the Release Note Required field to identify which release this deprecation will be going out in.
+1. Add a `O3DE_DEPRECATED(<GHI Number>)` comment right above the API call.
+    1. e.g.
+```c++
+    // O3DE_DEPRECATED(GHI-1234)
+    AZ::Entity* CreateEditorEntity(const char* name) = 0;
+```
+> Aside: To make searching easier, only add API comments to the function/type declarations, not definitions.
+1. Create second GHI to add `AZ_DEPRECATED` to the API.
+    1. Add the Deprecation label.
+    1. Set this issue to be Blocked by the first issue.
```

Comment:
GHI's can't be blocked, may need to rethink/reword this.

Reply:
Hmm good point, perhaps a new kind of label? Do you know who would be best to ping here to ask about this? Thanks!

Reply:
I would start with @mcphedar  or @forhalle .

Reply:
Fantastic thanks, I've dropped them a message :+1: